### PR TITLE
feat: implement ephemeral accounts

### LINKED ISSUE #914: feat: implement ephemeral accounts
Problem:
Creating state on Solana requires locking rent-exempt balance (3,480 lamports/byte). For a 10KB account, this costs ~35M lamports. This is prohibitively expensive for temporary use cases like gaming sessions, caching, or computation intermediate results.

Proposal:
Implement ephemeral accounts — zero-balance accounts with an EPHEMERAL flag that live only in the ephemeral rollup. Rent is paid by a sponsor account at 32 lamports/byte (~109x cheaper than Solana), with full refund on closure.

Three new instructions:
- CreateEphemeralAccount(data_len)
- ResizeEphemeralAccount(new_data_len)
- CloseEphemeralAccount

CPI-only architecture ensures programs mediate access. Rent flows through a vault account owned by magic-program.

Acceptance criteria:
- [x] EPHEMERAL flag added to AccountSharedData (bit 6)
- [x] Create/Resize/Close instructions implemented in magic-program
- [x] Sponsor validation (PDA or oncurve with proper signing/ownership)
- [x] Vault account initialized and blacklisted from base chain
- [x] Ephemeral accounts rejected in ScheduleCommit
- [x] Comprehensive test suite
- [ ] SDK helper methods and documentation (Phase 4)

### PR DESCRIPTION
**Pull request by bmuddha**
## Summary

Implements **ephemeral accounts** — zero-balance accounts that live only in the ephemeral rollup, with rent paid by a sponsor at **32 lamports/byte** (~109x cheaper than Solana).

This enables cheap temporary storage for gaming sessions, caching, computation intermediate results, etc. — without base chain prerequisites.

**What's included:**
- Three new instructions: `CreateEphemeralAccount`, `ResizeEphemeralAccount`, `CloseEphemeralAccount`
- CPI-only architecture (programs mediate access)
- Rent flows through a vault account, fully refundable on close
- Comprehensive test suite (~1,100 lines)

See [MIMD-0016](https://github.com/magicblock-labs/magicblock-validator/discussions/840) for full spec (rent analysis, safeguards, example usage).

## Compatibility
- [x] No breaking changes
- [ ] Config change (describe):
- [ ] Migration needed (describe):

## Testing
- [x] tests added (ephemeral_accounts.rs with coverage for create/resize/close flows)

## Checklist
- [ ] docs updated (if needed)
- [x] closes #914
- [x] depends on: https://github.com/magicblock-labs/solana-account/pull/29, https://github.com/magicblock-labs/magicblock-svm/pull/12

## Summary by CodeRabbit

* **New Features**
  * Ephemeral accounts: create, resize, and close low-cost temporary accounts; ephemeral vault funded at startup; sponsor-aware invocation support.

* **Validation & Safety**
  * New runtime checks to prevent improper use, mutation, or committing of ephemeral accounts; ownership and rent rules enforced.

* **Tests**
  * Extensive integration test suite covering lifecycle, edge cases, failure paths, and sponsor scenarios.

* **Chores**
  * Workspace version bumped to 0.7.0 and Solana dependency revisions updated; dev tooling package version bumped.

### CODE REVIEW

File: programs/magicblock/src/ephemeral_accounts/process_create.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        .ok_or(InstructionError::IncorrectProgramId)?;
+
+    // Validate sponsor (signer or PDA owned by caller)
+    validate_sponsor(transaction_context)?;
+
+    // Validate vault is owned by magic program
+    let vault =
+        accounts::get_instruction_account_with_idx(transaction_context, 2)?;
+    if *vault.borrow().owner() != id() {
+        return Err(InstructionError::InvalidAccountOwner);
```
[the remark this answers is not in the document]
Comment (bmuddha) [reply programs/magicblock/src/ephemeral_accounts/process_create.rs]: DONE

File: programs/magicblock/src/ephemeral_accounts/process_create.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    if data_len > MAX_DATA_LEN {
+        return Err(InstructionError::InvalidArgument);
+    }
+
+    let caller_program_id = validate_common(transaction_context)?;
+    validate_ephemeral_signer(transaction_context)?;
+    let ephemeral = validate_new_ephemeral(transaction_context)?;
+
+    let rent = rent_for(data_len)?;
+    transfer_rent(transaction_context, rent as i64)?;
```
[the remark this answers is not in the document]
Comment (bmuddha) [reply programs/magicblock/src/ephemeral_accounts/process_create.rs]: impossible to overflow in nearest decade
